src/components/CustomCollection/CustomCollection.tsx

```diff
@@ -295,38 +324,71 @@ const CustomCollection = ({
     setEditingTitle(false)
   }
 
-  const handleOnDragEnd = (result: DropResult) => {
-    const { source, destination } = result
-
-    if (destination) {
-      let copiedBookmarks = visibleBookmarks.map(
-        ({ _id, url, label, cmsId, isRemoved }) => ({
-          _id,
-          url,
-          label,
-          cmsId,
-          isRemoved,
-        })
-      )
+  const handleOnDragStart = (event) => {
+    setActiveDragId(event.active.id)
+  }
+
+  const handleOnDragEnd = async (event) => {
+    setActiveDragId(null)
+    const { active, over } = event
+
+    // If a draggable item is active, and it is over a droppable area when dropped
+    if (over && active.id !== over.id) {
+      setDragAndDropSortableItems((items) => {
+        const oldIndex = items.indexOf(active.id)
+        const newIndex = items.indexOf(over.id)
 
-      const [removed] = copiedBookmarks.splice(source.index, 1)
-      copiedBookmarks.splice(destination.index, 0, removed)
+        // dnd-kit sorts the items for us when we call arrayMove when an item is dropped
+        const sortedItems = arrayMove(items, oldIndex, newIndex)
 
-      if (removedBookmarks.length > 0) {
-        const removedBookmarksToAddBack = removedBookmarks.map(
-          ({ _id, url, label, cmsId, isRemoved }) => ({
-            _id,
-            url,
-            label,
-            cmsId,
-            isRemoved,
-          })
-        )
+        // This is to update the visibleBookmarks array to match the order of the sortedItems array. We can then perform
+        // a mutation to update the order of the bookmarks in the database.
+        for (let i = 0; i < sortedItems.length; i++) {
+          if (sortedItems[i] !== visibleBookmarks[i]._id.toString()) {
+            const sortedId = sortedItems[i]
 
-        copiedBookmarks = [...copiedBookmarks, ...removedBookmarksToAddBack]
-      }
+            const sortedBookmark = visibleBookmarks.find(
+              (b) => b._id.toString() === sortedId
+            ) as MongoBookmark
 
-      handleEditCollection(title, copiedBookmarks)
+            const sortedBookmarkIndex = visibleBookmarks.findIndex(
+              (b) => b._id.toString() === sortedId
+            )
+
+            const sortedBookmarkCopy = { ...sortedBookmark }
+            const visibleBookmarksCopy = [...visibleBookmarks]
+
+            visibleBookmarksCopy.splice(sortedBookmarkIndex, 1)
+            visibleBookmarksCopy.splice(i, 0, sortedBookmarkCopy)
+            setVisibleBookmarks(visibleBookmarksCopy)
+          }
+        }
+        return sortedItems
+      })
```

Comment:
I hope the comment I left makes sense. But if not, here is a short summary: dnd-kit has an extension that allows for sorting of a list of draggable items. In order to use that sorting functionality, we need to pass an array of unique ids (unfortunately, it has to be an array of strings or something similar, can't just use the bookmark objects), for which I used the _id of each bookmark. When you drop a draggable item, the previously mentioned array is updated in state to represent the new order of the items. We also need to update the order of `visibleBookmarks` because it contains all of the actual bookmark details, and we need to use it in our mutation to update what is in the db.